File: project files/server/src/models/ImportJob.js
const mongoose = require('mongoose');

const importJobSchema = new mongoose.Schema(
  {
    jobType: { type: String, enum: ['PRODUCT_CSV', 'STOCK_CSV'], required: true, index: true },
    status: { type: String, enum: ['PENDING', 'RUNNING', 'COMPLETED', 'FAILED'], default: 'PENDING', index: true },
    createdBy: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
    totalRows: { type: Number, default: 0 },
    processedRows: { type: Number, default: 0 },
    successRows: { type: Number, default: 0 },
    failedRows: { type: Number, default: 0 },
    failureReport: {
      type: [
        {
          rowNumber: { type: Number, required: true },
          reason: { type: String, required: true }
        }
      ],
      default: []
    },
    notificationSent: { type: Boolean, default: false },
    startedAt: { type: Date, default: null },
    completedAt: { type: Date, default: null },
    errorMessage: { type: String, default: '' }
  },
  { timestamps: true }
);

importJobSchema.index({ createdBy: 1, status: 1, createdAt: -1 });

module.exports = mongoose.model('ImportJob', importJobSchema);

File: spring_back/src/main/java/com/groceryapp/model/ImportJob.java
package com.groceryapp.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "import_jobs")
public class ImportJob {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    @Column(name = "job_type")
    private String jobType;
    @Column(name = "status")
    private String status;
    @Column(name = "created_by")
    private String createdBy;
    private Integer totalRows;
    private Integer processedRows;
    private Integer successRows;
    private Integer failedRows;
    @Builder.Default
    private List<Failure> failureReport = new ArrayList<>();
    private Boolean notificationSent;
    private Instant startedAt;
    private Instant completedAt;
    private String errorMessage;
    private Instant createdAt;
    private Instant updatedAt;

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Failure {
        private Integer rowNumber;
        private String reason;
    }
}
